src/HotChocolate/Language/src/Language.SyntaxTree/Utilities/SyntaxSerializer.QuerySyntax.cs

```diff
@@ -0,0 +1,224 @@
+
+                if (_indented)
+                {
+                    writer.WriteLine();
+                    writer.Unindent();
+                }
+                else
+                {
+                    writer.WriteSpace();
+                }
+
+                writer.WriteIndent();
+                writer.Write('}');
+            }
+        }
+
+        private void VisitSelection(ISelectionNode node, ISyntaxWriter context)
+        {
+            switch (node.Kind)
+            {
+                case NodeKind.Field:
+                    VisitField((FieldNode)node, context);
+                    break;
+                case NodeKind.FragmentSpread:
+                    VisitFragmentSpread((FragmentSpreadNode)node, context);
+                    break;
+                case NodeKind.InlineFragment:
+                    VisitInlineFragment((InlineFragmentNode)node, context);
+                    break;
+                default:
+                    throw new NotSupportedException();
+            }
+        }
+
+        private void VisitField(FieldNode node, ISyntaxWriter writer)
+        {
+            writer.WriteIndent();
+
+            if (node.Alias is { })
+            {
+                writer.WriteName(node.Alias);
+                writer.Write(": ");
+            }
+
+            writer.WriteName(node.Name);
+
+            if (node.Arguments.Count > 0)
+            {
+                writer.Write('(');
+                writer.WriteMany(node.Arguments, (n, w) => w.WriteArgument(n));
+                writer.Write(')');
+            }
+
+            if (node.Directives.Count > 0)
+            {
+                writer.WriteSpace();
+                writer.WriteMany(node.Directives,
+                    (n, w) => w.WriteDirective(n),
+                    w => w.WriteSpace());
+            }
+
+            if (node.SelectionSet is { } selectionSet && selectionSet.Selections.Count > 0)
+            {
+                writer.WriteSpace();
+                VisitSelectionSet(selectionSet, writer);
+            }
+        }
+
+        private void VisitFragmentSpread(FragmentSpreadNode node, ISyntaxWriter writer)
+        {
+            writer.WriteIndent();
+
+            writer.Write("... ");
+            writer.WriteName(node.Name);
+
+            if (node.Directives.Count > 0)
+            {
+                writer.WriteMany(node.Directives,
+                    (n, w) => w.WriteDirective(n),
+                    w => w.WriteSpace());
+            }
+        }
+
+        private void VisitInlineFragment(InlineFragmentNode node, ISyntaxWriter writer)
+        {
+            writer.WriteIndent();
+
+            writer.Write("...");
+
+            if (node.TypeCondition is { })
+            {
+                writer.WriteSpace();
+                writer.Write(Keywords.On);
+                writer.WriteSpace();
+
+                writer.WriteNamedType(node.TypeCondition);
+            }
+
```

Comment:
would this be no TypeCondition be allowed? Should we fail?

Reply:
This is a fragment definition and it must have a type condition. On inline fragments the type condition can be omitted since in this case the inline fragment would inherit the type condition